larq_compute_engine/core/indirect_bgemm/kernel_8x4x2_aarch64.h

```diff
@@ -91,10 +88,49 @@ namespace kernel_8x4x2_aarch64 {
   "uadalp %[accum_3].8h, v3.16b\n\t"
 
 /**
- * This function performs the accumulation loop when we know that the depth is
- * greater than one, i.e. the (bitpacked) number of input channels is greater
- * than 64.
+ * Accumulation loops
+ *
+ * There are two variants of the accumulation loop: one for when we know the
+ * depth is greater than one, i.e. the number of input channels is greater than
+ * 64; and one for when we know the depth is equal to one, i.e. the number of
+ * input channels is equal to 64. The latter case allows for a slight
+ * optimisation.
+ *
+ * The accumulation loops use inline assembly but are equivalent to the
+ * following pseudocode:
+ *
+ *     accum_0 = 0;
+ *     accum_1 = 0;
+ *     accum_2 = 0;
+ *     accum_3 = 0;
+ *
+ *     // This block is removed in the depth=1 case
+ *     a_ptr_0 = indirection_buffer[0];
+ *     a_ptr_1 = indirection_buffer[1];
+ *     a_ptr_2 = indirection_buffer[2];
+ *     a_ptr_3 = indirection_buffer[3];
+ *     indirection_buffer += 4;
```

Comment:
Same suggestion as above
```suggestion
 *     // This block is removed in the depth=1 case
 *     // The first set of activations is already loaded, so this +1
 *     // ensures that the first 'block' loads the next set of activations.
 *     a_ptr_0 = indirection_buffer[0] + 1;
 *     a_ptr_1 = indirection_buffer[1] + 1;
 *     a_ptr_2 = indirection_buffer[2] + 1;
 *     a_ptr_3 = indirection_buffer[3] + 1;
 *     indirection_buffer += 4;
```